js/models/BaseModel.js

```diff
@@ -264,8 +325,185 @@ export default class extends Model {
+   *   in the currency module validateCurrencyAmount function.
+   * @param {function} addError - A function that takes the errKey and error string and
+   *   adds it to the model's errObj. This function is duplicated by most models in
+   *   their validate method. (there is a TODO to centralize this somewhere rather
+   *   than each validate having to duplicate it)
+   * @param {string} errKey - The error key corresponding to the field being validated.
+   *   This will be passed into the above addError function in the case of an error
+   *   being found.
+   * @param {object} options
+   * @param {object} options.translations - An object mapping error types to the
+   *   transalation keys that should be used when adding the error strings via addErr.
+   *   The translations have defaults, so only override if you need a specific error
+   *   message for a particular error or errors.
+   * @param {object} options.validationOptions - These options will be passed in as the
+   *   options to be used for validateCurrencyAmount from the currency module.
+   * @returns {object} - The return value of validateCurrencyAmount from the currency
+   *   module will be returned.
+   */
+  validateCurrencyAmount(
+    curDef,
+    addError,
+    errKey,
+    options = {}
+  ) {
+    if (typeof errKey !== 'string' || !errKey) {
+      throw new Error('The errKey must be provided as a non-empty string.');
+    }
+
+    if (typeof addError !== 'function') {
+      throw new Error('addError must be provided as a function.');
+    }
+
+    const opts = {
+      ...options,
+      translations: {
+        coinDiv: 'currencyAmountErrors.invalidCoinDiv',
+        required: 'currencyAmountErrors.missingValue',
+        type: 'currencyAmountErrors.invalidType',
+        fractionDigitCount: 'currencyAmountErrors.fractionTooLow',
+        ...options.translations,
+      },
+    };
+
+    // This should match the default range in validateCurrencyAmount()
+    const defaultRange = CUR_VAL_RANGE_TYPES.GREATER_THAN_ZERO;
+
+    if (!opts.translations.range) {
+      const range =
+        options.validationOptions &&
+        options.validationOptions.rangeType ||
+        defaultRange;
+
+      if (range === CUR_VAL_RANGE_TYPES.GREATER_THAN_ZERO) {
+        opts.translations.range = 'currencyAmountErrors.greaterThanZero';
+      } else if (range === CUR_VAL_RANGE_TYPES.GREATER_THAN_OR_EQUAL_ZERO) {
+        opts.translations.range = 'currencyAmountErrors.greaterThanEqualZero';
+      }
+    }
+
+    let divisibility;
+    let currency;
+
+    try {
+      divisibility = _.result(curDef.currency, 'divisibility');
+      currency = _.result(curDef.currency, 'code');
+    } catch (e) {
+      // pass
+    }
+
+    const validation = validateCurrencyAmount(
+      curDef.amount,
+      divisibility,
+      opts.validationOptions
+    );
+
+    if (
+      validation.validCoinDiv === false &&
+      typeof opts.translations.coinDiv === 'string'
```

Comment:
That was by design because if the caller doesn't want a particular error to be added to the error object, they could pass in `false` for that translation:

I updated the docs:

```
   * @param {object} options.translations - An object mapping error types to the
   *   transalation keys that should be used when adding the error strings via addErr.
   *   The translations have defaults, so only override if you need a specific error
   *   message for a particular error or errors. Pass in false if you don't want a
   *   particular error added to the error object.
```

I don't remember the specifics, but I did have a use-case where I needed to use that.